
<launch>
    
    <arg name="model" />
    <arg name="rviz_visualize" default="true" />
    <param name="robot_description" textfile="$(find g1_description)/urdf/g1.urdf" />

    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen">
      <param name="publish_frequency" value="1000"/>
      <param name="use_tf_static"     value="true"/>
  </node>

    <node if="$(arg rviz_visualize)" 
        pkg="rviz" 
        type="rviz" 
        name="rviz" 
        args="-d $(find mocap)/config/replay.rviz" 
        output="screen" />

    <node pkg="tf" type="static_transform_publisher" name="world"
      args="0 0 0 0 0 0 map world 100" />

    <node name="replay_visualize" pkg="mocap" type="replay_vis.py" output="screen"/>
    
    <param name="replay_fps" value="120" />
    <param name="Interpolation" value="True" />
    <param name="wrist_motion" value="False" />
    <param name="extend_link" value="True" />

    <param name="start_frame" value="200" />
    <param name="end_frame" value="430" />   <!-- 终止帧，-1为默认长度 -->
    <param name="root_height_offset" value="-0.36"/>
    <param name="outputdata" value="True" />

    
</launch>